### PR TITLE
feat(price-feeds): add uSPAC10g price feed

### PR DESCRIPTION
**Pull request by VasilyGerrans**
See: https://github.com/UMAprotocol/UMIPs/pull/559

Signed-off-by: VasilyGerrans <vgerrans@gmail.com>

### CODE REVIEW

File: packages/financial-templates-lib/src/price-feed/USPAC10GPriceFeed.ts [hunk 1]
```diff
@@ -0,0 +1,364 @@
...
+
+  private async fetchSymbolArray(verbose = false): Promise<string[]> {
+    const url = "https://spachero-spac-database.p.rapidapi.com/top10/";
+    const options = {
+      method: "GET",
+      headers: {
+        "x-rapidapi-host": "spachero-spac-database.p.rapidapi.com",
+        "x-rapidapi-key": this.rapidApiKey,
+      },
+      body: '{ period: "weekly", type: "common", sortby: "gainers" }',
```
Comment (abg4) [inline packages/financial-templates-lib/src/price-feed/USPAC10GPriceFeed.ts:332]: @VasilyGerrans I am getting the following error: `TypeError: Request with GET/HEAD method cannot have body`. When I comment out the body the response is returned but wanted to check if it works for you?

I added the config below to https://github.com/UMAprotocol/protocol/blob/master/packages/financial-templates-lib/src/price-feed/DefaultPriceFeedConfigs.ts#L1019 and included my rapidApiKey in my `PRICE_FEED_CONFIG` as an env variable.

`uSPAC10g: {
    type: "uSPAC10g",
    lookback: 7200
  }`

Comment (VasilyGerrans) [reply packages/financial-templates-lib/src/price-feed/USPAC10GPriceFeed.ts:332]: > @VasilyGerrans I am getting the following error: `TypeError: Request with GET/HEAD method cannot have body`. When I comment out the body the response is returned but wanted to check if it works for you?
> 
> I added the config below to https://github.com/UMAprotocol/protocol/blob/master/packages/financial-templates-lib/src/price-feed/DefaultPriceFeedConfigs.ts#L1019 and included my rapidApiKey in my `PRICE_FEED_CONFIG` as an env variable.
> 
> `uSPAC10g: { type: "uSPAC10g", lookback: 7200 }`

Could you also share the exact command you use for testing?

Comment (abg4) [reply packages/financial-templates-lib/src/price-feed/USPAC10GPriceFeed.ts:332]: Here is the command I am running (I have tried used various timestamps):

`HARDHAT_NETWORK=mainnet ./src/local/getHistoricalPrice.js --network mainnet_mnemonic --identifier uSPAC10g --time 1673276770 --ancillaryData 0x`

Comment (abg4) [reply packages/financial-templates-lib/src/price-feed/USPAC10GPriceFeed.ts:332]: Hey @VasilyGerrans, following up to see how it is looking on your end when you run the script or you had any questions.

**Comment by smb2796** [comment]
Closing due to inactivity

**Pull request closed without merging**